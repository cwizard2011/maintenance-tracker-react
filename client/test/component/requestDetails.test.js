import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import DefaultRequestDetails, { RequestDetails } from '../../src/components/RequestDetails';

const mockStore = configureMockStore([thunk]);

const props = {
  fetchSingleRequest: jest.fn(),
  loading: false,
  match: {
    params: {
      requestId: 'aa9-vv-33e'
    }
  },
  request: {
    created_at: '2018-10-30T13:32:54.578Z',
    currentstatus: 'pending',
    details: 'testing now',
    request_id: 'b50b13ef-f8f8-4b99-a33c-0d3a159f320b',
    title: 'Test request',
    updated_at: '2018-10-30T13:32:54.578Z',
    user_id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1'
  }
};
const initialState = {
  requestReducer: {
    request: {
      created_at: '2018-10-30T13:32:54.578Z',
      currentstatus: 'pending',
      details: 'testing now',
      request_id: 'b50b13ef-f8f8-4b99-a33c-0d3a159f320b',
      title: 'Test request',
      updated_at: '2018-10-30T13:32:54.578Z',
      user_id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1'
    },
    loading: false,
  }
};
const store = mockStore(initialState);

describe('Edit Request', () => {
  it('render pending requests correctly', () => {
    const wrapper = shallow(<RequestDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render connected component', () => {
    const wrapper = shallow(<DefaultRequestDetails store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
  it('should render resolved request correctly', () => {
    const newProps = {
      ...props,
      request: {
        created_at: '2018-10-30T13:32:54.578Z',
        currentstatus: 'resolved',
        details: 'testing now',
        request_id: 'b50b13ef-f8f8-4b99-a33c-0d3a159f320b',
        title: 'Test request',
        updated_at: '2018-10-30T13:32:54.578Z',
        user_id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1'
      }
    };
    const wrapper = shallow(<RequestDetails {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render resolved request correctly', () => {
    const newProps = {
      ...props,
      request: {
        created_at: '2018-10-30T13:32:54.578Z',
        currentstatus: 'approved',
        details: 'testing now',
        request_id: 'b50b13ef-f8f8-4b99-a33c-0d3a159f320b',
        title: 'Test request',
        updated_at: '2018-10-30T13:32:54.578Z',
        user_id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1'
      }
    };
    const wrapper = shallow(<RequestDetails {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render resolved request correctly', () => {
    const newProps = {
      ...props,
      request: {
        created_at: '2018-10-30T13:32:54.578Z',
        currentstatus: 'rejected',
        details: 'testing now',
        request_id: 'b50b13ef-f8f8-4b99-a33c-0d3a159f320b',
        title: 'Test request',
        updated_at: '2018-10-30T13:32:54.578Z',
        user_id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1'
      }
    };
    const wrapper = shallow(<RequestDetails {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should display loading component', () => {
    const newProps = {
      ...props,
      request: {},
      loading: true,
    };
    const wrapper = shallow(<RequestDetails {...newProps} />);
    expect(wrapper.find('Loading')).toBeDefined();
  });
  it('should display loading component', () => {
    const newProps = {
      ...props,
      request: {},
      loading: false,
    };
    const wrapper = shallow(<RequestDetails {...newProps} />);
    expect(wrapper.state()).toEqual(null);
  });
});
