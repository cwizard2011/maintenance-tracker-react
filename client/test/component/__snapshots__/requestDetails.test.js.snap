// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Request render pending requests correctly 1`] = `
<div>
  <Connect(UserNavigation) />
  <div
    className="container"
  >
    <div
      className="request"
    >
      <RequestDetailsContent
        created="Oct 30, 2018"
        details="testing now"
        labelClass="label pending"
        labelName="pause"
        requestId="b50b13ef-f8f8-4b99-a33c-0d3a159f320b"
        status="pending"
        title="Test request"
        updated="Oct 30, 2018"
      />
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Edit Request should render resolved request correctly 1`] = `
<div>
  <Connect(UserNavigation) />
  <div
    className="container"
  >
    <div
      className="request"
    >
      <RequestDetailsContent
        created="Oct 30, 2018"
        details="testing now"
        labelClass="label success"
        labelName="build"
        requestId="b50b13ef-f8f8-4b99-a33c-0d3a159f320b"
        status="resolved"
        title="Test request"
        updated="Oct 30, 2018"
      />
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Edit Request should render resolved request correctly 2`] = `
<div>
  <Connect(UserNavigation) />
  <div
    className="container"
  >
    <div
      className="request"
    >
      <RequestDetailsContent
        created="Oct 30, 2018"
        details="testing now"
        labelClass="label success"
        labelName="done-all"
        requestId="b50b13ef-f8f8-4b99-a33c-0d3a159f320b"
        status="approved"
        title="Test request"
        updated="Oct 30, 2018"
      />
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Edit Request should render resolved request correctly 3`] = `
<div>
  <Connect(UserNavigation) />
  <div
    className="container"
  >
    <div
      className="request"
    >
      <RequestDetailsContent
        created="Oct 30, 2018"
        details="testing now"
        labelClass="label danger"
        labelName="close"
        requestId="b50b13ef-f8f8-4b99-a33c-0d3a159f320b"
        status="rejected"
        title="Test request"
        updated="Oct 30, 2018"
      />
    </div>
  </div>
  <Footer />
</div>
`;
